src/GraphQl/Resolver/Stage/ReadStage.php

```diff
@@ -140,9 +140,14 @@ private function getNormalizedFilters(array $args): array
 
         foreach ($filters as $name => $value) {
             if (\is_array($value)) {
-                if (strpos($name, '_list')) {
+                if (\is_string($name) && strpos($name, '_list')) {
                     $name = substr($name, 0, \strlen($name) - \strlen('_list'));
                 }
+
+                // If it's just a simple array of string values, for example when using the _list syntax we don't want to decompose the array because it's will return multiple string and not multipe array that we can merge
```

Comment:
I think you should invert the sense of this sentence to tell the developer when this `if` is applied.
Like:
If the value contains arrays, we need to merge them for the filters to understand this syntax, proper to GraphQL to preserve the order of the arguments.